lib/index.js

```diff
@@ -191,12 +191,16 @@ CloudFront.prototype.request = function(method, path, headers, cb) {
 
       cb(err, info);
     });
-    parser.on("error", function(err) {
-      cb(err, info);
+    parser.on('error', function(err) {
+      cb(err);
     });
 
   });
 
+  req.on('error', function(err) {
```

Comment:
Yeah, just that the callback might be called twice. But you may be right that you will never see both an `response` and `error` event.